console/src/view/table.rs

```diff
@@ -0,0 +1,183 @@
+            Char('i') => {
+                let state = self.state_mut();
+                state.sort_descending = !state.sort_descending
+            }
+            Down => self.scroll_next(),
+            Up => self.scroll_prev(),
+            _ => {} // do nothing for now...
+        }
+        if let Ok(sort_by) = S::try_from(self.state().selected_column) {
+            let state = self.state_mut();
+            state.sort_by = sort_by;
+        }
+    }
+
+    fn scroll_with(&mut self, f: impl Fn(&Vec<Weak<RefCell<R>>>, usize) -> usize) {
+        // If the list of sorted items is empty, don't try to scroll...
+        let state = self.state_mut();
+        if state.sorted_items.is_empty() {
+            state.table_state.select(None);
+            return;
+        }
+
+        // Increment the currently selected row, or if no row is selected, start
+        // at the first row.
+        let i = state.table_state.selected().unwrap_or(0);
+        let i = f(&state.sorted_items, i);
+        state.table_state.select(Some(i));
+    }
+
+    fn scroll_next(&mut self) {
+        self.scroll_with(|resources, i| {
+            if i >= resources.len() - 1 {
+                // If the last itsm is currently selected, wrap around to the
+                // first item.
+                0
+            } else {
+                // Otherwise, increase the selected index by 1.
+                i + 1
+            }
+        });
+    }
+
+    fn scroll_prev(&mut self) {
+        self.scroll_with(|resources, i| {
+            if i == 0 {
+                // If the first item is currently selected, wrap around to the
+                // last item.
+                resources.len() - 1
+            } else {
+                // Otherwise, decrease the selected item by 1.
+                i - 1
+            }
+        })
+    }
+
+    fn selected_item(&self) -> Weak<RefCell<R>> {
+        let state = self.state();
+        state
+            .table_state
+            .selected()
+            .map(|i| {
+                let selected = if state.sort_descending {
+                    i
+                } else {
+                    state.sorted_items.len() - i - 1
+                };
+                state.sorted_items[selected].clone()
+            })
+            .unwrap_or_default()
+    }
+
+    fn controls(&self, styles: &view::Styles) -> Text {
+        tui::text::Text::from(Spans::from(vec![
+            Span::raw("controls: "),
+            bold(styles.if_utf8("\u{2190}\u{2192}", "left, right")),
+            text::Span::raw(" = select column (sort), "),
+            bold(styles.if_utf8("\u{2191}\u{2193}", "up, down")),
+            text::Span::raw(" = scroll, "),
+            bold(styles.if_utf8("\u{21B5}", "enter")),
+            text::Span::raw(" = view details, "),
+            bold("i"),
+            text::Span::raw(" = invert sort (highest/lowest), "),
+            bold("q"),
+            text::Span::raw(" = quit"),
+        ]))
+    }
+}
```

Comment:
nit: does this really need to be a method on `TableList`? it seems like it could just be a free function in this module, instead, since it never actually uses any data from `self`...

```suggestion
}

pub(in crate::view) fn controls(tyles: &view::Styles) -> Text {
    tui::text::Text::from(Spans::from(vec![
        Span::raw("controls: "),
        bold(styles.if_utf8("\u{2190}\u{2192}", "left, right")),
        text::Span::raw(" = select column (sort), "),
        bold(styles.if_utf8("\u{2191}\u{2193}", "up, down")),
        text::Span::raw(" = scroll, "),
        bold(styles.if_utf8("\u{21B5}", "enter")),
        text::Span::raw(" = view details, "),
        bold("i"),
        text::Span::raw(" = invert sort (highest/lowest), "),
        bold("q"),
        text::Span::raw(" = quit"),
    ]))
}
```